src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,89 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoError } from '../../types/TodoFilter';
+import { client } from '../../utils/fetchClient';
+
+type OneTodo = {
+  todo: Todo
+  todosUpdate: () => void
+  errorHandler: (errorType: TodoError) => void
+};
+
+export const TodoInfo: React.FC<OneTodo> = ({
+  todo,
+  todosUpdate,
+  errorHandler,
+}) => {
+  const [loader, setLoader] = useState(false);
+
+  const checkboxHandler = async () => {
+    setLoader(true);
+    errorHandler(TodoError.noerror);
+    const date = {
+      completed: !todo.completed,
+    };
+
+    try {
+      await client.patch(`/todos/${todo.id}`, date);
+      todosUpdate();
+    } catch {
+      errorHandler(TodoError.update);
+    } finally {
+      setLoader(false);
+    }
+  };
+
+  const buttonHandler = async () => {
+    setLoader(true);
```

Comment:
rename [boolean variables to better ones](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)